devito/passes/iet/mpi.py

```diff
@@ -108,6 +108,27 @@ def _merge_halospots(iet):
     Merge HaloSpots on the same Iteration tree level where all data dependencies
     would be honored.
     """
+
+    # Merge rules -- if the retval is True, then it means the input `dep` is not
+    # a stopper to halo merging
+
+    def rule0(dep, hs, *args):
+        # E.g., `dep=W<f,[t1, x]> -> R<f,[t0, x-1]>` => True
+        return not any(d in hs.dimensions or dep.distance_mapper[d] is S.Infinity
+                       for d in dep.cause)
+
+    def rule1(dep, *args):
+        # TODO This is apparently never hit, but feeling uncomfortable to remove it
```

Comment:
Bit scary but OK 😄

Reply:
this is in master. I tried to drop it and all tests still pass. I think `rule0`, which I introduced not so long ago, generalised rule1 plus the other rule that was here before it. I just hadn't thought about that until this PR. I prefer to stick it around for a little longer, but we can probably safely drop it